### ISSUE TITLE
Rename `typescript-all-in-one` pack to just `typescript`

### ISSUE THREAD

**Issue by 20k-ultra** [AstroNvim/astrocommunity#204]
I think that the existing `typescript` pack should become `typescript-node` since we have `typescript-deno`.

This than allows us to use `typescript` for the currently named `typescript-all-in-one` pack.

This will be a breaking change for people using the new pack but otherwise be backwards compatible for configs using `typescript` but now have deno!

cc @owittek

**Comment by 20k-ultra** [comment]
I could also see the argument to not do this to keep people's typescript configs more slim since someone developing typescript for node but never Deno will now have deno lsp.

**Comment by owittek** [comment]
Hey! There's a simple reason why I didn't change the name of the pack when we introduced the deno & all-in-one pack. It's because it would count as a breaking change and every current user of the pack would have to update the name which would lead to quite a lot of confusion as the error message in that case is also not too intuitive. Might be something to consider if we bump up the major version but that's up to @luxus & @mehalter to decide.

**Comment by 20k-ultra** [comment]
makes sense to me. I'm not convinced I like this idea anymore after I realized my above comment regarding bloat.

**Comment by mehalter** [comment]
yeah I'm not sure about how often this should allow breaking changes. The automated version release calculation will handle the version number bump correctly as long as the commit messages correctly indicate the breaking change with `!`. But I think for now, how it is is probably alright just to try and put off breaking changes for a bit

**Comment by mehalter** [comment]
Closing for now
